doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -476,106 +421,172 @@ Induction
-and an explanation of the underlying technique.
+   .. prodn::
+      simple_binder ::= @name
+      | ( {+ @name } : @term )
 
-.. seealso:: :tacn:`functional induction`
+   A primitive tactic to start a proof by induction. Generally,
+   higher-level tactics such as :tacn:`induction` or :tacn:`elim`
+   are easier to use.
 
-.. tacn:: discriminate @term
-   :name: discriminate
+   :n:`@ident` is the name of
+   the induction hypothesis. :n:`@natural` tells on which
+   premise of the current goal the induction acts, starting from 1,
+   counting both dependent and non-dependent products, but skipping local
+   definitions. The current lemma must be composed of at
+   least :n:`@natural` products.
 
-   This tactic proves any goal from an assumption stating that two
-   structurally different :n:`@term`\s of an inductive set are equal. For
-   example, from :g:`(S (S O))=(S O)` we can derive by absurdity any
-   proposition.
+   As in a fix expression, induction hypotheses must be used on
+   structurally smaller arguments. The verification that inductive proof
+   arguments are correct is done only when registering the
+   lemma in the global environment. To know if the use of induction hypotheses
+   is correct during the interactive development of a proof, use
+   the command :cmd:`Guarded` (see Section :ref:`requestinginformation`).
 
-   The argument :n:`@term` is assumed to be a proof of a statement of
-   conclusion :n:`@term = @term` with the two terms being elements of an
-   inductive set. To build the proof, the tactic traverses the normal forms
-   [1]_ of the terms looking for a couple of subterms :g:`u` and :g:`w` (:g:`u`
-   subterm of the normal form of :n:`@term` and :g:`w` subterm of the normal
-   form of :n:`@term`), placed at the same positions and whose head symbols are
-   two different constructors. If such a couple of subterms exists, then the
-   proof of the current goal is completed, otherwise the tactic fails.
+   :n:`with {+ ( @ident {* @simple_binder } {? %{ struct @name %} } : @type ) }`
+     Starts a proof by mutual induction. The statements to be proven
+     are :n:`forall @simple_binder__i, @type__i`.
+     The identifiers :n:`@ident` are the names of the induction hypotheses. The identifiers
+     :n:`@ident` are the respective names of the premises on which the induction
+     is performed in the statements to be proved (if not given, Coq
+     guesses what they are).
 
-.. note::
-   The syntax :n:`discriminate @ident` can be used to refer to a hypothesis
-   quantified in the goal. In this case, the quantified hypothesis whose name is
-   :n:`@ident` is first introduced in the local context using
-   :n:`intros until @ident`.
+.. tacn:: cofix @ident {? with {+ ( @ident {* @simple_binder } : @type ) } }
 
-.. exn:: No primitive equality found.
-   :undocumented:
+   .. todo:
 
-.. exn:: Not a discriminable equality.
-   :undocumented:
+   Starts a proof by co-induction. The first :n:`@ident` is the
```

Comment:
This "first" word is problematic since all the `ident` are the names of co-induction hypotheses but only the first one is provided if there is no `with` clause. So either use plural from the beginning, or apply this suggestion:

```suggestion
   Starts a proof by co-induction. The identifier :n:`@ident` is the
```

Reply:
OK?